src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TextMessageEncoder.cs

```diff
@@ -546,7 +546,9 @@ public override Task<ArraySegment<byte>> WriteMessageAsync(Message message, int
                     await message.WriteMessageAsync(xmlWriter);
                     xmlWriter.WriteEndDocument();
                 }
-                await xmlWriter.FlushAsync();
+
+                // The underlying type of xmlWriter, XmlUTF8TextWriter, has not implemented FlushAsync() yet.
+                xmlWriter.Flush();
```

Comment:
This is really crappy if we suddenly go sync while in a supposedly async path. Can you please open up a bug to track this implementation in the corefx repo?